client/src/pages/resources/components/resource-form.tsx

```diff
@@ -0,0 +1,316 @@
+import {
+  deleteResource,
+  fetchResource,
+  testResourceConnection,
+  updateResource,
+} from '../queries'
+import { Resource } from 'types/resource'
+
+export default function ResourceForm() {
+  const { resourceId } = useParams() as { resourceId: string }
+
+  const navigate = useNavigate()
+
+  const [form] = Form.useForm()
+
+  const { isLoading, data: resource, error } = useQuery(
+    ['resources', resourceId],
+    () => fetchResource(resourceId),
+    {
+      onSuccess: (data) => {
+        const { name, host, port, database, username, password } = data
+        form.setFieldsValue({ name, host, port, database, username, password })
+      },
+    },
+  )
+
+  const resources: Resource[] | undefined = queryCache.getQueryData([
+    'resources',
+  ])
+  const [disabled, setDisabled] = useState(true)
+
+  const [editResource] = useMutation(updateResource, {
+    onMutate: (updatedResource) => {
+      queryCache.setQueryData(
+        ['resources'],
+        resources?.map((resource) =>
+          resource.id === updatedResource.id
+            ? { ...resource, ...updatedResource }
+            : resource,
+        ),
+      )
+      queryCache.setQueryData(['resources', resourceId], {
+        ...resource,
+        ...updatedResource,
+      })
+    },
+    onSuccess: () => {
+      message.success('Resource updated successfully')
+    },
+  })
+
+  const [removeResource] = useMutation(deleteResource, {
+    onMutate: (deletedResource) => {
+      queryCache.setQueryData(
+        ['resources'],
+        resources?.filter((resource) => resource.id !== deletedResource.id),
+      )
+      queryCache.removeQueries(['resources', resourceId])
+      navigate('/resources')
+    },
+  })
+
+  const onFinish = useCallback(
+    (values: any) => {
+      const { name, host, port, database, username, password } = values
+      const id = parseInt(resourceId)
+      editResource({
+        id,
+        name,
+        host,
+        port,
+        database,
+        username,
+        password,
+      })
+    },
+    [editResource, resourceId],
+  )
+
+  const confirm = useCallback(() => {
+    Modal.confirm({
+      title: 'Delete this resource?',
+      content: 'This action cannot be reverted',
+      okText: 'Yes',
+      okType: 'danger',
+      cancelText: 'No',
+      onOk() {
+        removeResource({ id: parseInt(resourceId) })
+      },
+    })
+  }, [removeResource, resourceId])
+
+  const pingConnection = useCallback(async () => {
+    const {
+      name,
+      host,
+      port,
+      database,
+      username,
+      password,
+    } = form.getFieldsValue()
+    try {
+      const response = await testResourceConnection({
+        name,
+        type: 'postgres',
+        host,
+        port,
+        database,
+        username,
+        password,
+      })
+      message.success(response.message)
+    } catch (error) {
+      message.error(error.message)
+    }
+  }, [form])
+
+  const renderForm = useMemo(() => {
```

Comment:
Rename it to `formContent`. `renderForm` seems like a function which needs to be called to render form.